### PR TITLE
ICU-22714 fix zos build, use OpenXL

### PR DESCRIPTION
**Pull request by alexgubanow**
I have fixed the git clone of this repo on zos and moved build from old XLc compiler to new OpenXL compiler.
More details https://unicode-org.atlassian.net/browse/ICU-22714
As of now:
- tests are passing, some issues with files not being UTF-8 on zos, will address that in separate PR
- results of build was tested internally, we have own test suite
- following files had no BOM, need to fix tool that is used to generate those:
icu4c/source/data/misc/currencyNumericCodes.txt
icu4c/source/data/misc/icustd.txt
icu4c/source/data/misc/zoneinfo64.txt

##### Checklist

- [x] Required: Issue filed: https://unicode-org.atlassian.net/browse/ICU-22714
- [x] Required: The PR title must be prefixed with a JIRA Issue number. 
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. 
- [ ] Issue accepted (done by Technical Committee after discussion)
- [x] Tests included, if applicable
- [x] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

**Review by srl295 — APPROVED** [review]

**Comment by srl295** [comment]
lots of good deletion here!

For the BOMs, would it make sense just to fix them up with a build step?

**Comment by alexgubanow** [comment]
OpenXL is basically clang, that allowed me to get rid of many "OS390 only" code and tricks. Actually i have found one bug in OpenXL v2.1, it fails to generate correct name for sidedeck if output path is relative and library name has no suffix, for output library path like:
../../lib/LICU76DA
openxl v2.1 linker will generate ..x name of sidedeck
Fix for that, is already implemented, but no idea when it will be released by IBM, so i made a workaround, added "-Wl,-x$@.x" to linker command, so linker do not guess the name of sidedeck and no bug occur

I have added BOM fix in build time generated index files, also made sure those generated as UTF-8.

But regarding this files:
icu4c/source/data/misc/currencyNumericCodes.txt
icu4c/source/data/misc/icustd.txt
icu4c/source/data/misc/zoneinfo64.txt
i don't know when/who and how updates/generates it. Maybe you guys can guide me there? also i would suggest to do it in separate ticket/PR

**Comment by alexgubanow** [comment]
Hello @markusicu @srl295 what is missing for this PR to be merged?

**Review by eggrobin — APPROVED** [review]

**Comment by srl295** [comment]
@markusicu LGTM ready to merge?

**Comment by alexgubanow** [comment]
Hello @markusicu , can we merge this? is anything missing?

File: docs/userguide/icu4c/build.md [hunk 1]
```diff
@@ -368,55 +368,54 @@ Some platforms use package management tools to control the installation and unin
 
 ## How To Build And Install On z/OS (OS/390)
 
-You can install ICU on z/OS or OS/390 (the previous name of z/OS), but IBM tests only the z/OS installation. You install ICU in a z/OS UNIX system services file system such as HFS or zFS. On this platform, it is important that you understand a few details:
+You can install ICU on z/OS or OS/390 (the previous name of z/OS), but IBM tests only the z/OS installation. You install ICU in a z/OS UNIX System Services (z/OS UNIX) file system such as HFS or zFS. On this platform, it is important that you understand a few details:
```
Comment (markusicu) [inline docs/userguide/icu4c/build.md:371]: FYI: IBM hasn't testing anything for a long time... :-}

**Review by markusicu — APPROVED** [review]
Sorry for dropping the ball here! Let's get this in.
Happy to see that there are no merge conflicts.
Fingers crossed that the post-merge CI checks will still be green...

**Comment by markusicu** [comment]
I had to rebase in order to make a newer CI check work. Now we find out if it still works with the latest checks, even pre-merge...

**Pull request merged**